### DISCUSSION TITLE
Weird problem with importing going_modular

### DISCUSSION THREAD

**Discussion by mfarhadirad** [mrdbourke/pytorch-deep-learning#678] [Q&A] [answered]
hey Daniel
I got an error when I wrote this code. the only difference that this code has with yours is a space between going_modular and the dot after it. why does this happen? It says **No module named 'going_modular**. Your code works perfectly. I just want to know why we need a space there.

This is my code: 

```python
try:
  from going_modular.going_modular import data_setup, engine
except:
  print('Couldn`t find going madular scripts.. Downloading it...')
  !git clone https://github.com/mrdbourke/pytorch-deep-learning
  !mv pytorch-deep-learning/going_modular. # <--------- **This is the difference between mine and yours**
  !rm -rf pytorch-deep-learning
  from going_modular.going_modular import data_setup, engine
```

This is your code:

```python
try:
    from going_modular.going_modular import data_setup, engine
except:
    # Get the going_modular scripts
    print("[INFO] Couldn't find going_modular scripts... downloading them from GitHub.")
    !git clone https://github.com/mrdbourke/pytorch-deep-learning
    !mv pytorch-deep-learning/going_modular .
    !rm -rf pytorch-deep-learning
    from going_modular.going_modular import data_setup, engine
```

**Comment by mrdbourke** [comment, marked as answer]
Hi @mfarhadirad , 

Great question!

The `.` can be a little confusing.

Let me explain.

The `.` is short for "current directory".

For example, if I'm in the directory `daniel/` and I write `.`, it means I'm referring to `daniel/`.

In the line: 

```
!mv pytorch-deep-learning/going_modular .
```

It is saying "move the contents of `going_modular` to here".

As in, if the directory is laid out like:

```
daniel/
    pytorch-deep-learning/
        going_modular/
            file1.py
```

And I'm currently in `daniel/`, it will become:

```
daniel/
    file.py
    pytorch-deep-learning/
        going_modular/
```

Another way of thinking of it is:

```
!mv pytorch-deep-learning/going_modular .
```

Stands for: 

```
!mv source_files destination_files
```

So the contents of `pytorch-deep-learning/going_modular` is the `source_files` and `.` (here or the current directory) is the `destination_files`.

So you need the space in ` .` to differentiate the `source_files` and the `destination_files`.

-- 

Let me know if this helps!

See here for more: https://superuser.com/questions/171871/what-is-the-difference-between-and

**Reply by mfarhadirad** [reply]
Thank you so much for your clear answer.

**Reply by mrdbourke** [reply]
You're welcome!

**Reply by injamul3798** [reply]
Thanks a lot mrdbourke
